models/articles.models.js

```diff
@@ -11,19 +11,49 @@ exports.fetchArticleById = (article_id) => {
   });
 };
 
+exports.fetchAllArticles = (topic, sort_by = "created_at", order = "desc") => {
+  const acceptedSorts = [
+    "author",
+    "title",
+    "article_id",
+    "created_at",
+    "votes",
+    "article_img_url",
+    "comment_count",
+  ];
 
-exports.fetchAllArticles = () => {
-  const text =
-    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;";
-  return db.query(text).then(({ rows }) => {
+  const acceptedOrders = ["asc", "desc"];
+
+  if (!acceptedSorts.includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  if (!acceptedOrders.includes(order)) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+  let text = "";
+  const params = [];
+  text += `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id `;
+
+  if (topic) {
+    text += `WHERE topic = $1 `;
+    params.push(topic);
+  }
+
+  text += `GROUP BY articles.article_id ORDER BY ${sort_by} ${order};`;
+  return db.query(text, params).then(({ rows }) => {
+    if (rows.length===0){
```

Comment:
This is something to look at for the missing test I mentioned in your test suite.